### PR TITLE
APIKIT-15479: Hide Location/Position if no info returned from AMF.

### PR DESCRIPTION
**Pull request by nicolaskosacoff**

### CODE REVIEW

File: interface-impl-amf/src/main/java/org/mule/amf/impl/parser/rule/ApiValidationResultImpl.java [hunk 1]
```diff
@@ -35,9 +35,13 @@ public ApiValidationResultImpl(amf.client.validate.ValidationResult validationRe
 
   @Override
   public String getMessage() {
-    return buildErrorMessage(validationResult.message(),
-                             validationResult.location().orElse(""),
-                             validationResult.position().start());
+    String formattedErrorMessage =   format(validationResult.message());
+    Optional <String> location = validationResult.location();
+    Range positionRange = validationResult.position();
+    if(!location.empty().isPresent() && !positionRange.start().isZero()) {
```
Comment (dcourtil) [inline interface-impl-amf/src/main/java/org/mule/amf/impl/parser/rule/ApiValidationResultImpl.java:41]: ```location.empty()``` returns an empty optional, so ```!location.empty().isPresent()``` is always true. Get rid of ```empty()``` and the negation, leave just ```location.isPresent()``` to check if there is value for the optional

File: interface-impl-amf/src/main/java/org/mule/amf/impl/parser/rule/ApiValidationResultImpl.java [hunk 2]
```diff
@@ -35,9 +35,13 @@ public ApiValidationResultImpl(amf.client.validate.ValidationResult validationRe
 
   @Override
   public String getMessage() {
-    return buildErrorMessage(validationResult.message(),
-                             validationResult.location().orElse(""),
-                             validationResult.position().start());
+    String formattedErrorMessage =   format(validationResult.message());
+    Optional <String> location = validationResult.location();
```
Comment (dcourtil) [inline interface-impl-amf/src/main/java/org/mule/amf/impl/parser/rule/ApiValidationResultImpl.java:39]: It seems that the formatter is not working. Get rid of the space between ```Optional``` and ```<String>```

File: interface-impl-amf/src/test/java/org/mule/amf/impl/parser/rule/ApiValidationResultImplTest.java [hunk 3]
```diff
@@ -7,23 +7,34 @@
...
+import scala.Option;
 
 import java.util.List;
 import java.util.Optional;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertNull;
-import static org.junit.Assert.assertTrue;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.*;
```
Comment (dcourtil) [inline interface-impl-amf/src/test/java/org/mule/amf/impl/parser/rule/ApiValidationResultImplTest.java:26]: I would avoid wildcard imports.

File: interface-impl-amf/src/main/java/org/mule/amf/impl/parser/rule/ApiValidationResultImpl.java [hunk 4]
```diff
@@ -35,9 +35,13 @@ public ApiValidationResultImpl(amf.client.validate.ValidationResult validationRe
 
   @Override
   public String getMessage() {
-    return buildErrorMessage(validationResult.message(),
-                             validationResult.location().orElse(""),
-                             validationResult.position().start());
+    String formattedErrorMessage =   format(validationResult.message());
```
Comment (JuanAller) [inline interface-impl-amf/src/main/java/org/mule/amf/impl/parser/rule/ApiValidationResultImpl.java:38]: why using String.format without arguments ?

File: interface-impl-amf/src/main/java/org/mule/amf/impl/parser/rule/ApiValidationResultImpl.java [hunk 5]
```diff
@@ -35,9 +35,13 @@ public ApiValidationResultImpl(amf.client.validate.ValidationResult validationRe
...
+    Optional<String> location = validationResult.location();
+    Range positionRange = validationResult.position();
+    if(location.isPresent() && !positionRange.start().isZero()) {
+      formattedErrorMessage = format(ERROR_FORMAT, validationResult.message(), URLDecoder.decode(location.get()), getPositionMessage(positionRange.start()));
```
Comment (JuanAller) [inline interface-impl-amf/src/main/java/org/mule/amf/impl/parser/rule/ApiValidationResultImpl.java:42]: return format(ERROR_FORMAT, validationResult.message(), URLDecoder.decode(location.get()), getPositionMessage(positionRange.start()));

File: interface-impl-amf/src/main/java/org/mule/amf/impl/parser/rule/ApiValidationResultImpl.java [hunk 6]
```diff
@@ -35,9 +35,13 @@ public ApiValidationResultImpl(amf.client.validate.ValidationResult validationRe
...
+    }
+    return formattedErrorMessage;
```
Comment (JuanAller) [inline interface-impl-amf/src/main/java/org/mule/amf/impl/parser/rule/ApiValidationResultImpl.java:44]: return validationResult.message()

File: interface-impl-amf/src/test/java/org/mule/amf/impl/parser/rule/ApiValidationResultImplTest.java [hunk 7]
```diff
@@ -62,4 +73,36 @@ public void getSeverityTest() {
...
+        ValidationResult validationResult = createAMFValidationResult(message, start, location);
+        ApiValidationResultImpl apiValidationResult = new ApiValidationResultImpl(validationResult);
+
+        String actualMessage = apiValidationResult.getMessage();
+        String expectedMessage = FULL_CUSTOM_ERROR_MESSAGE;
+        assertThat(actualMessage, is(expectedMessage));
+    }
+
+    @Test
+    public void testWithoutDetails() throws Exception{
```
Comment (JuanAller) [inline interface-impl-amf/src/test/java/org/mule/amf/impl/parser/rule/ApiValidationResultImplTest.java:91]: add test where location is not empty, and position is 0,0 , for full condition coverage

File: interface-impl-amf/src/test/java/org/mule/amf/impl/parser/rule/ApiValidationResultImplTest.java [hunk 8]
```diff
@@ -7,23 +7,34 @@
...
-import static org.junit.Assert.assertNull;
 import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertNull;
 
 public class ApiValidationResultImplTest {
 
     private static final String NO_VALID_RESOURCE_PATH = "not-valid-resource";
     private static final String ERROR_MESSAGE = "expected type: Integer, found: Double";
+    public static final String CUSTOM_ERROR_MESSAGE = "This is an error message!";
+    public static final String CUSTOM_LOCATION = "Here is the error";
```
Comment (dcourtil) [inline interface-impl-amf/src/test/java/org/mule/amf/impl/parser/rule/ApiValidationResultImplTest.java:34]: Can we add a URL to try the URLDecoder as well? Just to have an example of how it looks like
